apoc.cypher.runFile fails when there is a empty transaction

## Guidelines
## Expected Behavior (Mandatory)

apoc.cypher.runFile should be able to tolerate an empty txn


## Actual Behavior (Mandatory)
stored proc fails, for example

neo4j> call apoc.cypher.runFile('/home/neo4j/neo4j-enterprise-3.5.5/import/chain.cleanup.cypher');

Failed to invoke procedure `apoc.cypher.runFile`: Caused by: org.neo4j.cypher.internal.v3_5.util.SyntaxException: Invalid input ':': expected <init> (line 2, column 1 (offset: 1))


## How to Reproduce the Problem
Create a file, named runFile.cyp with the following content

```
:begin
:commit
````

attempt to use apoc.cypher.runFile();


## Specifications (Mandatory)

Currently used versions

### Versions

  - OS:  Ubuntu 18
  - Neo4j:  3.5.5
  - Neo4j-Apoc:  apoc-3.5.0.3.jar

--- a/src/main/java/apoc/cypher/Cypher.java
+++ b/src/main/java/apoc/cypher/Cypher.java
@@ -172,7 +172,9 @@ public class Cypher {
     private String removeShellControlCommands(String stmt) {
         Matcher matcher = shellControl.matcher(stmt.trim());
         if (matcher.find()) {
-            stmt = matcher.replaceAll("");
+            // an empty file get transformed into ":begin\n:commit" and that statement is not matched by the pattern
+            // because ":begin\n:commit".replaceAll("") => "\n:commit" with the recursion we avoid the problem
+            return removeShellControlCommands(matcher.replaceAll(""));
         }
         return stmt;
     }

--- a/src/test/java/apoc/cypher/CypherTest.java
+++ b/src/test/java/apoc/cypher/CypherTest.java
@@ -457,4 +457,10 @@ public class CypherTest {
                     assertEquals("C", ((Map) r.get("value")).get("cName"));
                 });
     }
+
+    @Test
+    public void testRunFileWithEmptyFile() throws Exception {
+        testResult(db, "CALL apoc.cypher.runFile('src/test/resources/empty.cypher')",
+                r -> assertFalse("should be empty", r.hasNext()));
+    }
 }

--- a/src/test/resources/empty.cypher
+++ b/src/test/resources/empty.cypher
@@ -1,0 +1,2 @@
+:begin
+:commit

